sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -286,7 +289,8 @@ public Diagnostic GetRuleViolation()
                 var mostGeneralType = FindMostGeneralType();
 
                 if (!Equals(mostGeneralType, parameterSymbol.Type) &&
-                    ShouldReportOnType(mostGeneralType.GetSymbolType()))
+                    IsConsistentAccessibility(mostGeneralType.GetEffectiveAccessibility(), methodAccessibility) &&
```

Comment:
This is subtly wrong. First, we find the most general type we can use, then we see if it is accessible. The correct logic would be to apply this logic at every step when finding most general type we can use.

I don't think it's worth changing, but it would be good to add a test, or at least a comment,